client/src/App.js

```diff
@@ -1,11 +1,135 @@
+
+  useEffect(() => {
+    fetch("https://videos-1rb3.onrender.com/")
+      .then((response) => response.json())
+      .then((data) => {
+        setVideos(data);
+        setLoading(false);
+    })
+      .catch((error) => console.log(error));
+  }, [updated]);
+
+  const addVideo = (newVideo) => {
+    fetch("https://videos-1rb3.onrender.com/", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(newVideo),
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log("Response:", data);
+        setUpdated(!updated);
+      })
+      .catch((error) => {
+        console.error("Error:", error);
+      });
+  };
+
+  const upVote = (video) => {
+    const updatedVideos = videos.map((v) =>
+      v === video ? { ...v, rating: v.rating + 1 } : v
+    );
+    setVideos(updatedVideos);
+    
+    fetch(`https://videos-1rb3.onrender.com/update_rating`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ id: video.id, rating: video.rating + 1 }),
+    })
+      .then((response) => response.json())
+      .then((data) => console.log(data))
+      .catch((error) => console.log("Error: ", error));
+  }
+
+  const downVote = (video) => {
+    const updatedVideos = videos.map((v) =>
+      v === video ? { ...v, rating: v.rating - 1 } : v
+    );
+    setVideos(updatedVideos);
+    fetch(`https://videos-1rb3.onrender.com/update_rating`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ id: video.id, rating: video.rating - 1 }),
+    })
+      .then((response) => response.json())
+      .then((data) => console.log(data))
+      .catch((error) => console.log("Error: ", error));
+  };
+
+  const removeVideo = (video) => {
+    fetch(`https://videos-1rb3.onrender.com/${video.id}`, {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log("Response:", data);
+        setUpdated(!updated);
+      })
+      .catch((error) => {
+        console.error("Error:", error);
+      });
+  };
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <h1>Video Recommendation</h1>
+    <div className="container-fluid">
+      <header className="header">
+        <h1>Video Recommendations</h1>
       </header>
+      <div className="control-panel">
+        <AddingItem addVideo={addVideo} />
+        <button
+          className="btn btn-warning"
+          onClick={() => setSortOrder(sortOrder === "desc" ? "asc" : "desc")}
+        >
+          Toggle Sort Order
+        </button>
+      </div>
+
+      <ErrorBoundary
+        fallback={<div>Could not connect to the server, sorry....</div>}
+      >
+        {loading && "Loading..."}
```

Comment:
To write a more readable code, we can move this part above the return and write it as below

```
if(isloading){
  return <>Loading...</>
}


```

then you dont need !loading checking on line 112